### ISSUE TITLE
Automatically validate the email verification code (no need to press <ENTER>)

### ISSUE THREAD

**Issue by Grraahaam** [Infisical/infisical#1582]
### Feature description

Today, when I create an account, I'm receiving the verification code to sign in by email.  I then need to copy and paste it in the app, after what I need to press <kbd>ENTER</kbd> or the submit button.

1. We could validate the code directly when pasting it in the input, or once all the fields have been populated?
2. I've seen a topic somewhere here about replacing the verification code by a link to simplify this process

What do you think?

### Why would it be useful?

It'd improve the UX and simplify the user's verification process

### Additional context

Related issues:
- https://github.com/Infisical/infisical/issues/58
- https://github.com/Infisical/infisical/pull/683
- https://github.com/Infisical/infisical/pull/1529
- https://github.com/Infisical/infisical/issues/555

**Comment by kavinda1995** [comment]
@Grraahaam I can work on this issue if anyone else is not working currently

**Comment by DanielHougaard** [comment]
@Grraahaam I agree, that would be even better. I wouldn't start doing that just yet though. We need to make the emails nicer in general at some point, and I think such a change would fall pretty nicely into that scope. 

For now I think making it automatically continue efter entering the code is the right move!

**Comment by DanielHougaard** [comment]
Oh wow I didn't see that one, awesome! React email is great, and this is actually what I had in mind for the mail refactor. Currently it's not a _huge_ priority, but it will defiantly become a more pressing topic in the future.

**Comment by vishavsingla** [comment]
Is this issue resolved?

**Comment by siddhant-agrawal01** [comment]
Is the issue resolved?

**Comment by akshat99812** [comment]
I'll take this if it's available

**Comment by rahuljagtapp** [comment]
@Grraahaam I can work on this issue  if issue is open

**Comment by saloni0419** [comment]
@Grraahaam Can you please assign me this issue, if it is still open

**Comment by Bhaveshshah777** [comment]
@DanielHougaard Here is a pull request to solve this bug , #2977 . Hope this works

**Comment by Ananmayd** [comment]
@Grraahaam I would love to work on this issue if it is still open.

**Comment by Siddharth-2382** [comment]
@DanielHougaard I've made the changes and created a PR. I'd appreciate it if someone could review it and get it merged.

**Comment by Definisi** [comment]
Makes sense, no reason to require an extra click/enter when the code is already complete.

Opened a fix in #5801 that auto-submits once all digits are filled in. Added a useEffect that watches the code length and triggers the existing submit handler when it matches the expected length (6 for email/TOTP, 8 for recovery codes). The verify button still works as a fallback.

Applied across all the code input screens: signup email verification, login MFA, SSO email confirmation, and the standalone MFA session page.

**Comment by Sanjays2402** [comment]
Hey — I'd like to take a crack at this. Planning to submit a PR in the next couple days. Will comment back if I hit anything unexpected.

**Comment by derinbarutcu17** [comment]
Hey! I'd like to implement this. My plan: detect when all OTP input fields are populated (onChange event on the last field), then call the submit handler directly — no Enter key needed. Works both for manual typing and paste. Is there a preferred UX for paste (split across fields vs. single input)?

**Comment by LeonxLJX** [comment]
Hi! I'd like to work on this issue. Please let me know if I can be assigned. Thanks!

**Comment by LeonxLJX** [comment]
Hi! I'd like to automatically validate the email verification code without pressing the button. Plan: add auto-submit on code completion, add tests. May I be assigned?
